pagerduty/resource_pagerduty_event_orchestration_path_router.go

```diff
@@ -136,56 +140,65 @@ func resourcePagerDutyEventOrchestrationPathRouterRead(d *schema.ResourceData, m
 		return nil
 	})
 
+	if retryErr != nil {
+		return diag.FromErr(retryErr)
+	}
+
+	return diags
 }
 
 // EventOrchestrationPath cannot be created, use update to add / edit / remove rules and sets
-func resourcePagerDutyEventOrchestrationPathRouterCreate(d *schema.ResourceData, meta interface{}) error {
-	return resourcePagerDutyEventOrchestrationPathRouterUpdate(d, meta)
+func resourcePagerDutyEventOrchestrationPathRouterCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	return resourcePagerDutyEventOrchestrationPathRouterUpdate(ctx, d, meta)
 }
 
-func resourcePagerDutyEventOrchestrationPathRouterDelete(d *schema.ResourceData, meta interface{}) error {
+func resourcePagerDutyEventOrchestrationPathRouterDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+
 	d.SetId("")
-	return nil
+	return diags
 }
 
-func resourcePagerDutyEventOrchestrationPathRouterUpdate(d *schema.ResourceData, meta interface{}) error {
+func resourcePagerDutyEventOrchestrationPathRouterUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+
 	client, err := meta.(*Config).Client()
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
 
-	updatePath := buildRouterPathStructForUpdate(d)
-
-	log.Printf("[INFO] Updating PagerDuty Event Orchestration Path of type %s for orchestration: %s", "router", updatePath.Parent.ID)
+	routerPath := buildRouterPathStructForUpdate(d)
+	var warnings []*pagerduty.EventOrchestrationPathWarning
 
-	return performRouterPathUpdate(d, updatePath, client)
-}
+	log.Printf("[INFO] Updating PagerDuty Event Orchestration Path of type %s for orchestration: %s", "router", routerPath.Parent.ID)
 
-func performRouterPathUpdate(d *schema.ResourceData, routerPath *pagerduty.EventOrchestrationPath, client *pagerduty.Client) error {
```

Comment:
This function was used in one spot and the code seemed more legible if I pasted it directly inside the `resourcePagerDutyEventOrchestrationPathRouterUpdate` so that's what I did.